elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/dimension/EntityDimension.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.dimension;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.datastores.aggregation.Column;
+import com.yahoo.elide.datastores.aggregation.annotation.Cardinality;
+import com.yahoo.elide.datastores.aggregation.annotation.CardinalitySize;
+import com.yahoo.elide.datastores.aggregation.annotation.FriendlyName;
+import com.yahoo.elide.datastores.aggregation.annotation.Meta;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+/**
+ * A dimension backed by a table.
+ * <p>
+ * {@link EntityDimension} is thread-safe and can be accessed by multiple threads.
+ */
+@Slf4j
+public class EntityDimension extends Column implements Dimension {
+
+    /**
+     * Returns the entity field that is defined to be a human displayable column of that entity.
+     *
+     * @param cls  The entity or a relation
+     * @param entityDictionary  A object that helps to find all fields with {@link FriendlyName} annotation
+     *
+     * @return friendlyName of the entity
+     *
+     * @throws IllegalStateException if more than 1 fields are annotated by the {@link FriendlyName}
+     */
+    public static String getFriendlyNameField(Class<?> cls, EntityDictionary entityDictionary) {
+        List<String> singleFriendlyName = entityDictionary.getAllFields(cls).stream()
+                .filter(field -> entityDictionary.attributeOrRelationAnnotationExists(
+                        cls,
+                        field,
+                        FriendlyName.class
+                ))
+                .collect(Collectors.toList());
```

Comment:
This can be simplified like this:
```java
      entityDictionary.getAllFields(cls).stream()
                .filter(field -> entityDictionary.attributeOrRelationAnnotationExists(
                        cls,
                        field,
                        FriendlyName.class
                ))
               .reduce((a, b) -> {
                     throw new IllegalStateException("Multiple @FriendlyName fields: " + a + ", " + b);
               })
              .get();
```